jwst/dark_current/tests/test_dark_sub.py

```diff
@@ -0,0 +1,517 @@
+        gend_ind = gstrt_ind + nframes
+
+        # Prepare arrays to hold results of averaging
+        manual_avg = np.zeros((ngroups))
+        manual_errs = np.zeros((ngroups))
+
+        # Manually average the input data to compare with pipeline output
+        for newgp, gstart, gend in zip(range(ngroups), gstrt_ind, gend_ind):
+
+            # Average the data frames
+            newframe = np.mean(dark.data[gstart:gend, 500, 500])
+            manual_avg[newgp] = newframe
+
+            # ERR arrays will be quadratic sum of error values
+            manual_errs[newgp] = np.sqrt(np.sum(dark.err[gstart:gend, 500, 500]**2)) / (gend - gstart)
+
+        # Check that pipeline output matches manual averaging results
+        assert np.all(manual_avg == avg_dark.data[:, 500, 500])
+        assert np.all(manual_errs == avg_dark.err[:, 500, 500])
+
+        # Check that meta data was properly updated
+        assert avg_dark.meta.exposure.nframes == nframes
+        assert avg_dark.meta.exposure.ngroups == ngroups
+        assert avg_dark.meta.exposure.groupgap == groupgap
+
+
+
+def test_more_sci_frames():
+    '''Check that data is unchanged if there are more frames in the science data is than in the reference file'''
+
+    # size of integration
+    nints = 1
+    ngroups = 30
+    xsize = 1032
+    ysize = 1024
+
+    # create raw input data for step
+    dm_ramp = make_rampmodel(nints, ngroups, ysize, xsize)
+    dm_ramp.meta.exposure.nframes = 1
+    dm_ramp.meta.exposure.groupgap = 0
+
+    # populate data array of science cube
+    for i in range(0, ngroups-1):
+        dm_ramp.data[0, i, :, :] = i
+
+    refgroups = 20
+    # create dark reference file model with fewer frames than science data
+    dark = make_darkmodel(refgroups, ysize, xsize)
+
+    # populate data array of reference file
+    for i in range(0, refgroups - 1):
+        dark.data[0, i, :, :] = i * 0.1
+
+    # apply correction
+    outfile = darkcorr(dm_ramp, dark)
+
+    # check that no correction/subtraction was applied; input file = output file
+    diff = dm_ramp.data[:, :, :, :] - outfile.data[:, :, :, :]
+
+    # test that the science data are not changed
+
+    np.testing.assert_array_equal(np.full((nints, ngroups, ysize, xsize), 0.0, dtype=float),
+                                  diff, err_msg='no changes should be seen in array ')
+
+
+def test_sub_by_frame():
+    '''Check that if NFRAMES=1 and GROUPGAP=0 for the science data, the dark reference data are
+    directly subtracted frame by frame'''
+
+    # size of integration
+    nints = 1
+    ngroups = 30
+    xsize = 1032
+    ysize = 1024
+
+    # create raw input data for step
+    dm_ramp = make_rampmodel(nints, ngroups, ysize, xsize)
+    dm_ramp.meta.exposure.nframes = 1
+    dm_ramp.meta.exposure.groupgap = 0
+
+    # populate data array of science cube
+    for i in range(0, ngroups-1):
+        dm_ramp.data[0, i, :, :] = i
+
+    refgroups = 50
```

Comment:
As above, for this type of test it's not necessary to have realistic data array dimensions, and hence xsize, ysize, ngroups, and refgroups could be reduced, if these tests end up being a burden on the overall CI unit test runs. (Remember, every time anyone does a commit, all of these tests are going to run).